### PR TITLE
Use global.require to silence webpack warning

### PR DESCRIPTION
**Pull request by mischnic**
# ↪️ Pull Request

I'm still not quite sure whether module.require and require are completely equivalent. Doing `toString` in node show this:

```js
module.require = function(id) {
  validateString(id, 'id');
  if (id === '') {
    throw new ERR_INVALID_ARG_VALUE('id', id, 'must be a non-empty string');
  }
  return Module._load(id, this, /* isMain */ false);
};

global.require = function(path) {
  try {
    exports.requireDepth += 1;
    return mod.require(path);
  } finally {
    exports.requireDepth -= 1;
  }
};
```

Related: #2883

### CODE REVIEW

**Comment by mischnic** [comment]
Is `typeof global.require === "function"` needed as well?

**Comment by devongovett** [comment]
I don’t think require is actually a global in node right? It wraps modules in a closure with a require argument, just like parcel does. So checking global.require probably won’t work.

**Comment by mischnic** [comment]
You're right...
But `module.require` and `require` are slightly different.

**Comment by devongovett** [comment]
Should we check that both module.require and require exist, but actually call require?

**Comment by mischnic** [comment]
> but actually call require?

No, because the original issue is that with `typeof require === 'function' && require`, Webpack can infer that `nodeRequire` is `require` and tries to bundle that call.

File: packages/shared/scope-hoisting/src/prelude.js [hunk 1]
```diff
@@ -6,8 +6,8 @@ parcelRequire = function(name) {
   }
 
   // Try the node require function if it exists.
-  if (typeof require === 'function') {
-    return require(name);
+  if (typeof global !== 'undefined' && typeof global.require === 'function') {
```
Comment (devongovett) [inline packages/shared/scope-hoisting/src/prelude.js]: Probably should update this one as well 😉

File: packages/packagers/js/src/prelude.js [hunk 2]
```diff
@@ -10,7 +10,10 @@
 parcelRequire = function(modules, cache, entry, globalName) {
   // Save the require from previous bundle to this closure if any
   var previousRequire = typeof parcelRequire === 'function' && parcelRequire;
-  var nodeRequire = typeof require === 'function' && require;
+  var nodeRequire =
+    typeof module !== 'undefined' &&
+    typeof module.require === 'function' &&
+    module.require.bind(module);
```
Comment (devongovett) [inline packages/packagers/js/src/prelude.js]: Can you put a comment about why we don't just use `require` so someone doesn't try to change this later?

**Pull request merged**